agent/config/conditional.go

```diff
@@ -24,17 +24,61 @@ const (
 	_ Conditional = iota
 	ExplicitlyEnabled
 	ExplicitlyDisabled
-	DefaultEnabled
+	NotSet
 )
 
+type BooleanDefaultTrue struct {
```

Comment:
if the goal is to fix the merge of boolean config values, i'm wondering can we just change the config type from bool to *bool instead of having this struct (the merge can differentiate a nil pointer from a pointer that has false value).

Reply:
I considered that option, but felt that it has higher risk of backward incompatibility with previous agent versions.
We already had this "Conditional" struct as a solution for one of the config, so that's what I expanded. I am not opposed to the *bool solution, but it'll take some looking into, to make sure that it's feasible solution for backward compatibility as well.

Reply:
since I keep forgetting this, leaving in a comment - plain boolean pointer won't work for our case since we have at least 2 configs that defaults to true when not set.